var/spack/repos/builtin/packages/py-pylint/package.py

```diff
@@ -45,31 +46,33 @@ class PyPylint(PythonPackage):
+    depends_on("py-astroid@1.5.1:", when="@1.7:", type=("build", "run"))
+    depends_on("py-astroid@1.6:1.9", when="@1.9.4", type=("build", "run"))
+    depends_on("py-astroid@2.0:", when="@2.2.0:", type=("build", "run"))
+    depends_on("py-astroid@2.2.0:2", when="@2.3.0:2.7", type=("build", "run"))
+    depends_on("py-astroid@2.5.6:2.6", when="@2.8.0:2.10", type=("build", "run"))
+    depends_on("py-astroid@2.8.0:2.8", when="@2.11.1", type=("build", "run"))
+    depends_on("py-astroid@2.11.2:2.11", when="@2.13.5:2.13", type=("build", "run"))
+    depends_on("py-astroid@2.11.6:2.11", when="@2.14.2:2.14", type=("build", "run"))
+    depends_on("py-astroid@2.12.4:2.14", when="@2.15.0:", type=("build", "run"))
     depends_on("py-backports-functools-lru-cache", when="^python@:2.8", type=("build", "run"))
     depends_on("py-configparser", when="^python@:2.8", type=("build", "run"))
     depends_on("py-dill@0.2:", when="@2.13.5:", type=("build", "run"))
-    depends_on("py-editdistance", type=("build", "run"), when="@:1.7")
+    depends_on("py-editdistance", when="@:1.7", type=("build", "run"))
     depends_on("py-isort@4.2.5:", type=("build", "run"))
     depends_on("py-isort@4.2.5:5", when="@2.3.1:", type=("build", "run"))
     depends_on("py-mccabe", type=("build", "run"))
     depends_on("py-mccabe@0.6.0:0.6", when="@2.3.1:2.11", type=("build", "run"))
     depends_on("py-mccabe@0.6.0:0.7", when="@2.13:", type=("build", "run"))
     depends_on("py-pip", type=("build"))  # see https://github.com/spack/spack/issues/27075
     # depends_on('py-setuptools-scm@1.15.0:', type='build')
-    depends_on("py-setuptools-scm", type="build", when="@2.8.2")
+    depends_on("py-setuptools-scm", when="@2.8.2", type="build")
     depends_on("py-setuptools@17.1:", type="build")
+    depends_on("py-setuptools@62.6:62", when="@2.15.0:", type="build")
     depends_on("py-singledispatch", when="^python@:3.3", type=("build", "run"))
-    depends_on("py-six", type=("build", "run"), when="@1:2.3.1")
-    depends_on("py-toml@0.7.1:", type=("build", "run"), when="@2.8.2:2.12.2")
-    depends_on("py-tomli@1.1.0:", type=("build", "run"), when="@2.13.5: ^python@:3.10")
-    depends_on("py-tomlkit@0.10.1:", type=("build", "run"), when="@2.14.0:")
-    depends_on("py-platformdirs@2.2.0:", type=("build", "run"), when="@2.11.1:")
-    depends_on("py-typing-extensions@3.10.0:", type=("build", "run"), when="@2.11.1: ^python@:3.9")
+    depends_on("py-six", when="@1:2.3.1", type=("build", "run"))
+    depends_on("py-toml@0.7.1:", when="@2.8.2:2.12.2", type=("build", "run"))
+    depends_on("py-tomli@1.1.0:", when="@2.13.5: ^python@:3.10", type=("build", "run"))
+    depends_on("py-tomlkit@0.10.1:", when="@2.14.0:", type=("build", "run"))
+    depends_on("py-platformdirs@2.2.0:", when="@2.11.1:", type=("build", "run"))
+    depends_on("py-typing-extensions@3.10.0:", when="@2.11.1: ^python@:3.9", type=("build", "run"))
```

Comment:
This package would be easier to review if the deps are listed in the same order as `pyproject.toml`